src/app/plugins/crumbl.plugin.ts

```diff
@@ -0,0 +1,134 @@
+// Original bot code: https://github.com/rob-3/crumblbot
+
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IHttpResponse, IMessage, ChannelType, Maybe } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+
+export class CrumblPlugin extends Plugin {
+  public name: string = 'Crumbl Cookies Plugin';
+  public description: string = 'Returns the cookies available that week at Crumbl';
+  public usage: string = 'crumbl';
+  public pluginAlias = [];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+  public buildId: string = '';
+  private _lastPost: Maybe<IMessage> = null;
+  
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public _matchId (data: string, regex: RegExp): string
+  {
+    const matches = data.match(regex);
+
+    if (matches !== null && matches.length > 1)
+    {
+      return matches[1];
+    }
+    
+    return '';
+  }
```

Comment:
There is some magic going on here, maybe do some array destructuring, or at least some comments
Also, we prefer to check the negative and return
```
if(!matches.length){
    return '';
}
//Do stuff
```